internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1318,6 +1319,18 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 							Type:     pluginsdk.TypeBool,
 							Optional: true,
 						},
+						"revisions": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							Computed: true,
```

Comment:
Is there a reason this needs to be `Computed`? We avoid introducing new properties that are both `Optional`+`Computed` (O+C), in particular lists of primitives unless there is a good reason to. This is because the provider is sometimes unable to tell whether a change has happened to an O+C property when a user tries to update it. We have a more detailed explanation over in the [contributor docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#setting-properties-to-optional--computed) in case you're interested.

Reply:
Updated in latest commit